### PR TITLE
Cycle the Riccati solver

### PR DESCRIPTION
**Pull request by edantec**
This PR aims at adding a function to efficiently cycle the three different kind of linear solvers so as to avoid full reallocation at each MPC control cycle. Discussions are open to decide what is the best way to achieve it.

### CODE REVIEW

**Comment by ManifoldFR** [comment]
I'm thinking - maybe we should have a test, reproducing some minimal example? And maybe checking for memory allocations when we expect none.

**Comment by ManifoldFR** [comment]
Didn't see a test was added yet!

File: tests/cycling.cpp [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+#include "aligator/solvers/proxddp/solver-proxddp.hpp"
+#include "aligator/core/explicit-dynamics.hpp"
+#include "aligator/core/cost-abstract.hpp"
+#include "aligator/utils/mpc-util.hpp"
+#include <proxsuite-nlp/modelling/spaces/pinocchio-groups.hpp>
+#include <proxsuite-nlp/third-party/polymorphic_cxx14.hpp>
+#include <boost/test/unit_test.hpp>
+
+#include <fmt/format.h>
+#include <fmt/ostream.h>
```
Comment (ManifoldFR) [inline tests/cycling.cpp:11]: Are these includes needed?

File: gar/include/aligator/gar/parallel-solver.hxx [hunk 2]
```diff
@@ -196,13 +196,33 @@ bool ParallelRiccatiSolver<Scalar>::forward(
...
+  condensedFacs.ldlt.back() = Eigen::BunchKaufman<MatrixXs>(dim1);
+};
+
 template <typename Scalar>
 void ParallelRiccatiSolver<Scalar>::initializeTridiagSystem(
     const std::vector<long> &dims) {
   ALIGATOR_TRACY_ZONE_SCOPED;
-  std::vector<MatrixXs> subdiagonal;
-  std::vector<MatrixXs> diagonal;
-  std::vector<MatrixXs> superdiagonal;
```
Comment (ManifoldFR) [inline gar/include/aligator/gar/parallel-solver.hxx:205]: Good job finding this unused stuff. wtf was I doing ?

**Review by ManifoldFR — DISMISSED** [review]
Fantastic work @edantec, you the real mvp

**Comment by edantec** [comment]
The cycling of the parallel solver does not work yet. Test case only involves serial solver for now. The linear solver dedicated to parallel computation needs to be redesign.

**Review by ManifoldFR — APPROVED** [review]

**Pull request merged**